lib/ddtrace/span.rb

```diff
@@ -68,7 +68,11 @@ def get_tag(key)
     # Set the given key / value metric pair on the span. Keys must be string.
     # Values must be floating point numbers.
     def set_metric(key, value)
+      # enforce that the value is a floating point number
+      value = Float(value)
       @metrics[key] = value
+    rescue StandardError => e
```

Comment:
if `nil` is passed it will be a `TypeError`:
```
irb(main):001:0> Float(nil)
TypeError: can't convert nil into Float
	from (irb):1:in `Float'
	from (irb):1
	from /Users/emanuele/.rbenv/versions/2.3.1/bin/irb:11:in `<main>'
```
The reason to use the `Float(value)` is that a string may be converted to `float`, as we're doing in the Python implementation. I think it was already discussed before about why we should at least try to make it work.